src/HttpJsonRpcRequestFactory.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\JsonRpc;
+
+use Nyholm\Psr7\Stream;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use function array_filter;
+use function Safe\json_encode;
+use function uniqid;
+
+final class HttpJsonRpcRequestFactory implements JsonRpcRequestFactory
+{
+    /** @var RequestFactoryInterface */
+    private $messageFactory;
+
+    public function __construct(RequestFactoryInterface $messageFactory)
+    {
+        $this->messageFactory = $messageFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function notification(string $method, ?array $params = null) : RequestInterface
+    {
+        return $this->createRequest(
+            JsonRpcMethod::NOTIFICATION,
+            [
+                'jsonrpc' => self::SPEC,
+                'method'  => $method,
+                'params'  => $params,
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function request($id, string $method, ?array $params = null) : RequestInterface
+    {
+        return $this->createRequest(
+            JsonRpcMethod::REQUEST,
+            [
+                'jsonrpc' => self::SPEC,
+                'method'  => $method,
+                'params'  => $params,
+                'id'      => $id,
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createRequest(string $method, array $options = []) : RequestInterface
+    {
+        $body = json_encode(
```

Comment:
možná je to zbytečnost jo, ale co takhle ještě mít DataSerializer? Aby bylo možné posílat body v jiném formátu než v jsonu ... pro nás useless, pro ostatní možné ne ... just an idea

Reply:
zatim na to nemam usecase, tak bych na to kaslal. PRs accepted :D slo hlavne o to promazat tu puvodni lib https://github.com/graze/guzzle-jsonrpc a co nejvic zjednodusit

Reply:
ok netrvám na tom, jen mě to tak hned napadlo ... takhle cajk :)